ddtrace/tracer.py

```diff
@@ -249,52 +255,44 @@ def configure(
 
         self._dogstatsd_url = dogstatsd_url or self._dogstatsd_url
 
-        if writer:
-            self.writer = writer
-            # Ensure dogstatsd client has been created for the writer being configured
-            self.writer.dogstatsd = get_dogstatsd_client(self._dogstatsd_url)
-        elif any(x is not None for x in [hostname, port, uds_path, https, priority_sampling, sampler]):
-            if any(x is not None for x in [hostname, port, uds_path, https]):
-                # If any of the parts of the URL have updated, merge them with
-                # the previous writer values.
-                if hasattr(self, "writer") and isinstance(self.writer, AgentWriter):
-                    prev_url_parsed = compat.parse.urlparse(self.writer.agent_url)
-                else:
-                    prev_url_parsed = compat.parse.urlparse("")
-
-                if uds_path is not None:
-                    if hostname is None and prev_url_parsed.scheme == "unix":
-                        hostname = prev_url_parsed.hostname
-                    url = "unix://%s%s" % (hostname or "", uds_path)
-                else:
-                    if https is None:
-                        https = prev_url_parsed.scheme == "https"
-                    if hostname is None:
-                        hostname = prev_url_parsed.hostname or ""
-                    if port is None:
-                        port = prev_url_parsed.port
-                    scheme = "https" if https else "http"
-                    url = "%s://%s:%s" % (scheme, hostname, port)
-            elif hasattr(self, "writer") and isinstance(self.writer, AgentWriter):
-                # Reuse the URL from the previous writer if there was one.
-                url = self.writer.agent_url
+        if any(x is not None for x in [hostname, port, uds_path, https]):
+            # If any of the parts of the URL have updated, merge them with
+            # the previous writer values.
+            if hasattr(self, "writer") and isinstance(self.writer, AgentWriter):
```

Comment:
With the changes in the `__init__` method, `hasattr(self, "writer")` always evaluates to `True` now

Reply:
Good catch! I missed porting it from #2121